betty/render.py

```diff
@@ -40,20 +41,73 @@ def media_types(self) -> Sequence[MediaType]:
         """
 
     @abstractmethod
-    async def render_file(
+    async def render(
         self,
-        file_path: Path,
+        content: str,
+        media_type: MediaType,
         *,
+        data: Mapping[str, Any] | None = None,
         job_context: Context | None = None,
         localizer: Localizer | None = None,
-    ) -> Path:
+    ) -> str:
+        """
+        Render content.
         """
-        Render a single file.
 
-        :return: The file's new path, which may have been changed, e.g. a
-            renderer-specific extension may have been stripped from the end.
+    def make_copy_function(
```

Comment:
This function gives us the most headaches because it is a bit of a hacky integration between renderers (who no longer know about files), files, wrapped up in a single function just so we can reuse it 2-3 times.